tools/apiview/emitters/cadl-apiview/src/emitter.ts

```diff
@@ -0,0 +1,94 @@
+// See: https://cadlwebsite.z1.web.core.windows.net/docs/extending-cadl/emitters-basics
+
+import {
+  emitFile,
+  getServiceNamespace,
+  getServiceTitle,
+  getServiceVersion,
+  Namespace,
+  Program,
+  projectProgram,
+  resolvePath,
+} from "@cadl-lang/compiler";
+import { buildVersionProjections } from "@cadl-lang/versioning";
+import { ApiViewDocument } from "./apiview.js";
+import { ApiViewEmitterOptions } from "./lib.js";
+
+const defaultOptions = {
+  "output-file": "apiview.json",
+} as const;
+
+export interface ResolvedApiViewEmitterOptions {
+  "output-file": string;
+}
+
+export async function $onEmit(program: Program, emitterOptions?: ApiViewEmitterOptions) {
+  const options = resolveOptions(program, emitterOptions ?? {});
+  const emitter = createApiViewEmitter(program, options);
+  await emitter.emitApiView();
+}
+
+export function resolveOptions(
+  program: Program,
+  options: ApiViewEmitterOptions
+): ResolvedApiViewEmitterOptions {
+  const resolvedOptions = { ...defaultOptions, ...options };
+
+  return {
+    "output-file": resolvePath(
+      program.compilerOptions.outputPath ?? "./output",
+      resolvedOptions["output-file"]
+    ),
+  };
+}
+
+// TODO: Move this to Cadl compiler?
+export function getFullyQualifiedNamespace(ns: Namespace, suffix?: string): string {
```

Comment:
there is `program.checker.getNamespaceString` (though I think we should expose that as a standalone helper. Can you create an issue for it.

Reply:
Updated the logic and created the issue you requested:
https://github.com/microsoft/cadl/issues/1057